packages/picasso.js/src/core/chart-components/ref-line/refline.js

```diff
@@ -196,7 +196,8 @@ const refLineComponent = {
 
       if (line.scale) {
         let scale = this.chart.scale(line.scale);
-        return extend(line, { scale, position: scale(line.value) });
+        const position = scale.min() === scale.max() && line.value !== scale.min() ? NaN : scale(line.value);
```

Comment:
As evident by the unit-test you have added, this will output nodes that have `NaN` coordinates. Which only works using SVG renderer because of a fix I made few weeks ago. It's a pattern we should avoid using even tho it happens  to work.

So add a condition to the filter functions on line 222 and 230 where the nodes are removed if the position is `NaN`.